pkgs/tools/misc/starship/default.nix

```diff
@@ -0,0 +1,24 @@
+{ stdenv, fetchFromGitHub, rustPlatform, openssl, pkgconfig, libiconv, darwin }:
```

Comment:
Does anyone know if it is preferred to pass `Security` here instead of `darwin`?

Reply:
Every time you pass one of the inner frameworks as an argument it has to be assigned specifically in `all-packages.nix`, which is why I tend to pass the full `darwin`, which is taken care of by `callPackage`.

Reply:
So I can resolve this suggestion and leave it like it is?

Reply:
@risicle yes, this is why you often see declarations like this inside `all-packages.nix`:
```
  sit = callPackage ../applications/version-management/sit {
    inherit (darwin.apple_sdk.frameworks) CoreFoundation Security;
  };
```

I'm not a `darwin` person, and I haven't ever really looked at the documentation about working with `darwin` in nix, so I'm genuinely asking which is preferred in nixpkgs :man_shrugging:

Reply:
Right now, in majority cases we import the entire darwin, but maybe @aanderse has a valid point and we inherit the frameworks instead.

Reply:
Well I think I've adequately derailed this PR just about enough. Sorry @bbigras I didn't mean to use your PR to try and set new `darwin` policy... especially since I've never used the platform!

For future reference it would be nice to know if there is a preferred style. That should probably be a decision left up to people like @matthewbauer and @LnL7, with input from people like @lilyball.

Reply:
I've seen both, but I personally just take `darwin`, I don't see any benefit to requiring `all-packages.nix` to do in the inherit.